spm-monitor/src/main/java/com/sematext/spm/client/db/DbObservation.java

```diff
@@ -91,6 +98,23 @@ public void read(ObservationDefinitionConfig observationDefinition) throws Confi
     // readAttributeNameMappings(observationDefinition);
     readIgnoreElements(observationDefinition);
     readAcceptElements(observationDefinition);
+    
+    readRowIdColumns(observationDefinition);
+  }
+
+  private void readRowIdColumns(ObservationDefinitionConfig observationDefinition) {
+    if (observationDefinition.getRowIdColumns() != null) {
+      rowIdColumns = new FastList<String>();
+      for (String id : observationDefinition.getRowIdColumns().split(",")) {
```

Comment:
Maybe we can squash this block with some functional syntactic sugar:

```java
List<String> rowIdColumns = Arrays.stream(observationDefinition.getRowIdColumns().split(","))
                .map(String::trim)
                .filter(id -> !id.equals(""))
                .collect(Collectors.toList());
```

Reply:
Very gladly, but again old Java, this has to be compiled for 1.6 :)

Reply:
ouch! Missed the previous comment : )

Reply:
Indeed, ouch, we have some nasty code because of that :(